common/src/main/java/com/wynntils/features/tooltips/ItemCompareFeature.java

```diff
@@ -4,162 +4,634 @@
+
+        int slot = hoveredSlot.getContainerSlot();
+
+        // Selecting armor and accessory slots is not allowed
+        for (int i : InventoryAccessory.getSlots()) {
+            if (slot == i) return;
+        }
+        for (int i : Inventory.ALL_ARMOR_SLOTS) {
+            if (slot == i + Inventory.INVENTORY_SIZE) return;
+        }
 
         ItemStack itemStack = hoveredSlot.getItem();
-        Optional<GearItem> wynnItemOpt = Models.Item.asWynnItem(itemStack, GearItem.class);
+        if (itemStack.isEmpty()) return;
+
+        Optional<WynnItem> wynnItemOpt = Models.Item.getWynnItem(itemStack);
         if (wynnItemOpt.isEmpty()) return;
+        WynnItem wynnItem = wynnItemOpt.get();
+        if (!(wynnItem instanceof GearItem)) return;
 
-        if (comparedItem == itemStack) {
-            comparedItem = null;
+        if (selectedItems.containsKey(wynnItem)) {
+            selectedItems.remove(wynnItem);
         } else {
-            comparedItem = itemStack;
-            compareToEquipped = false;
+            selectedItems.put(wynnItem, itemStack);
+        }
+    }
+
+    private String getTag(String key) {
+        return I18n.get(key);
+    }
+
+    private void drawSelectionArc(PoseStack poseStack, ItemStack itemStack, int slotX, int slotY, boolean hotbar) {
+        Optional<WynnItem> wynnItemOpt = Models.Item.getWynnItem(itemStack);
+        if (wynnItemOpt.isEmpty()) return;
+        if (selectedItems.containsKey(wynnItemOpt.get())) {
+            RenderUtils.drawArc(poseStack, CommonColors.LIGHT_BLUE, slotX, slotY, hotbar ? 0 : 200, 1, 6, 8);
+        }
+    }
+
+    private List<Component> getWynnOrVanillaLines(
+            AbstractContainerScreen<?> screen, WynnItem wynnItem, ItemStack itemStack) {
+        List<Component> wynnTooltip = TooltipUtils.getWynnItemTooltip(itemStack, wynnItem);
+        return wynnTooltip.isEmpty() ? screen.getTooltipFromItem(McUtils.mc(), itemStack) : wynnTooltip;
+    }
+
+    private MutableComponent getPaddedComponent(String string, int tooltipWidth) {
+        Font font = FontRenderer.getInstance().getFont();
+        // This is not the exact center, but good enough
+        int count = Math.round((tooltipWidth - font.width(string)) / 2.0f / font.width(" "));
+        return Component.literal(" ".repeat(count) + string);
+    }
+
+    private void removeFlavourText(List<Component> lines) {
+        TextColor loreColor = TextColor.fromLegacyFormat(ChatFormatting.DARK_GRAY);
+        for (int i = lines.size() - 1; i >= 0; --i) {
+            Component line = lines.get(i);
+
+            // Since line has dark_purple as color we get Component with actual color value with this
+            Component comp = StyledText.fromComponent(line).getComponent();
+            // This Component is empty, the Component we need is stored as it's sibling
+            List<Component> siblings = comp.getSiblings();
```

Comment:
Sorry, but I'd still like you to change this up a little bit. Basically, use StyledText to get the first "StyledTextPart", and then turn it into a component, and check that color. StyledText does not gurantee that the resulting components always looks like (empty -> allchildren), it just guarantees that the output's style matches visually/in meaning what was passed in.